CodePush.js

```diff
@@ -412,10 +413,14 @@ if (NativeCodePush) {
     restartApp,
     setUpTestDependencies,
     sync,
+    disallowRestart: RestartManager.disallow,
+    allowRestart: RestartManager.allow,
+    restartAllowed: RestartManager.allowed,
     InstallMode: {
       IMMEDIATE: NativeCodePush.codePushInstallModeImmediate, // Restart the app immediately
       ON_NEXT_RESTART: NativeCodePush.codePushInstallModeOnNextRestart, // Don't artificially restart the app. Allow the update to be "picked up" on the next app restart
-      ON_NEXT_RESUME: NativeCodePush.codePushInstallModeOnNextResume // Restart the app the next time it is resumed from the background
+      ON_NEXT_RESUME: NativeCodePush.codePushInstallModeOnNextResume, // Restart the app the next time it is resumed from the background
+      ON_NEXT_RESTART_OPPORTUNITY: NativeCodePush.codePushInstallModeOnNextRestartOpportunity
```

Comment:
I'll implement that.

If I remove the new install mode there is another potentially confusing issue: if I disable restarts and then receive a mandatory update it will also be delayed. I'm not sure if this is actually an issue but I could imagine the case that I want normal updates as soon as a restart is allowed but mandatory updates should still force an immediate restart. For my project it's not a problem, I'm not sure what other users need.